src/main/java/io/cdap/plugin/http/common/BaseHttpConfig.java

```diff
@@ -208,6 +223,19 @@ public String getOAuth2Enabled() {
         return oauth2Enabled;
     }
 
+    public OAuthGrantType getOauth2GrantType() {
+        OAuthGrantType grantType = OAuthGrantType.getGrantType(oauth2GrantType);
+        return BaseHttpSourceConfig.getEnumValueByString(OAuthGrantType.class, grantType.getValue(),
+            PROPERTY_OAUTH2_GRANT_TYPE);
+    }
+
+    public OAuthClientAuthentication getOauth2ClientAuthentication() {
+        OAuthClientAuthentication clientAuthentication = OAuthClientAuthentication.getClientAuthentication(
+            oauth2ClientAuthentication);
+        return BaseHttpSourceConfig.getEnumValueByString(OAuthClientAuthentication.class,
+            clientAuthentication.getValue(), PROPERTY_OAUTH2_CLIENT_AUTHENTICATION);
+    }
```

Comment:
Also both of them are new fields which means they can be `null` in existing pipelines, looks like it can throw `InvalidConfigPropertyException` in such cases.